packages/xo-web/src/xo-app/sr/tab-disks.js

```diff
@@ -123,67 +123,76 @@ const COLUMNS = [
         vms: getAllVms(state, props),
         vbds: getVbds(state, props),
       })
-    })(({ vbds, vms }) => {
-      if (isEmpty(vms)) {
+    })(({ item: vdi, vbds, vms, userData: { vmsSnapshotsBySuspendedVdi } }) => {
+      const vmSnapshot = vmsSnapshotsBySuspendedVdi[vdi.uuid]?.[0]
+      if (isEmpty(vms) && vmSnapshot === undefined) {
         return null
       }
 
       return (
         <Container>
-          {map(vbds, (vbd, index) => {
-            const vm = vms[vbd.VM]
+          {vbds.length > 0 ? (
```

Comment:
Why do you test this? Is it a test for suspend VDIs?

Reply:
Exactly, when i tried to console.log, VBDs was empty

Reply:
What about normal VDIs that don't have VBDs?

Isn't it where we want to use the `vmSnapshot === undefined` test, actually?